### ISSUE TITLE
Running Find-Module causes segmentation fault

### ISSUE THREAD

**Issue by quonic** [PowerShell/PowerShell#1921]
## Steps to reproduce

Installed latest release on Debian 8 - Jessie
`sudo apt-get install libunwind8 libicu52`

`sudo dpkg -i powershell_6.0.0-alpha.9-1ubuntu1.14.04.1_amd64.deb`

`powershell`

`Find-Module`
## Expected behavior

Return a list of modules.
## Actual behavior

ssh session closes as if I typed exit.
## Environment data

```
Name                           Value
----                           -----
PSVersion                      6.0.0-alpha
PSEdition                      Core
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   3.0.0.0
GitCommitId                    v6.0.0-alpha.9
CLRVersion
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by SteveL-MSFT** [comment]
@quonic can you include in the repro steps to establish the ssh session?

**Comment by quonic** [comment]
@SteveL-MSFT For the reproducible steps, I think upgrading from Debian 8.5 to 8.6 has fix sshd from closing the connection, but I've tested on some other machines to be sure.

The original machine that this happened was on a Linode VPS instance. I've just tested on it again after updating the system packages that it depends on and I get:
```
PS /root/downloads> Find-Module
Segmentation fault
root@linode:~/#
```

I've installed the latest release `powershell_6.0.0-alpha.10-1ubuntu1.14.04.1_amd64.deb` and I still get the Segmentation fault as above. The Segmentation fault might have been the cause of the ssh session to terminate on Debian 8.5.

I've also installed Debian 8.6 on a spare laptop and installed the latest release `powershell_6.0.0-alpha.10-1ubuntu1.14.04.1_amd64.deb` on it. and that gets the same Segmentation fault.

I used gdb to get the Seg fault:
```
PS /etc> [New Thread 0x7fff56db4700 (LWP 8354)]
[New Thread 0x7fff55db2700 (LWP 8355)]
[Thread 0x7ffff00cf700 (LWP 8345) exited]
Find-Module
[Thread 0x7fff56db4700 (LWP 8354) exited]
[New Thread 0x7fff56db4700 (LWP 8368)]
[New Thread 0x7ffff00cf700 (LWP 8369)]
[Thread 0x7ffff00cf700 (LWP 8369) exited]
[New Thread 0x7ffff00cf700 (LWP 8372)]
[Thread 0x7ffff00cf700 (LWP 8372) exited]

Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff56db4700 (LWP 8368)]
0x00007ffff09632ad in SystemNative_EnumerateInterfaceAddresses () from /opt/microsoft/powershell/6.0.0-alpha.10/System.Native.so
```

If that helps at all.

Thanks!

**Comment by lamw** [comment]
I'm also seeing a similiar behavior as described by @quonic when using latest Powershell Alpha14 on Debian 8.6 system. However, my issue is trying to load a VMware Module (PowerCLI Core). which then immediately seg faults. Interestingly, this had worked on an initial installation of Debian which I had blown away and re-built using the exact same instructions (which I've documented here http://www.virtuallyghetto.com/2017/01/how-to-install-powercli-core-on-debian-linux.html)

Debian Version (8.6)
```
root@deploy:~# uname -a
Linux deploy 3.16.0-4-amd64 #1 SMP Debian 3.16.36-1+deb8u2 (2016-10-19) x86_64 GNU/Linux
```

Powershell Version
```
PS /root> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.0.0-alpha
PSEdition                      Core
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   3.0.0.0
GitCommitId                    v6.0.0-alpha.14
CLRVersion
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

Here's output when trying to load module:

```
PS /root> Import-Module PowerCLI.Vds
PS /root> Segmentation fault
```

**Comment by thrnio** [comment]
I'm seeing similar behavior with alpha18 on Debian 9 "Stretch" (testing) using the Ubuntu 16.04 package, and for both `Find-Module` and `Update-Help`. Either cmdlet produces a segmentation fault with no output.

~~~
$ powershell
PowerShell 
Copyright (C) Microsoft Corporation. All rights reserved.

PS /home/rw> update-help                                                                                                                                                                                                                       
Segmentation fault

$ powershell
PowerShell 
Copyright (C) Microsoft Corporation. All rights reserved.

PS /home/rw> find-module                                                                                                                                                                                                                       
Segmentation fault

$ powershell
PowerShell 
Copyright (C) Microsoft Corporation. All rights reserved.

PS /home/rw>  $PSVersionTable                                                                                                                                                                                                                  

Name                           Value                                                                                                                                                                                                          
----                           -----                                                                                                                                                                                                          
PSVersion                      6.0.0-alpha                                                                                                                                                                                                    
PSEdition                      Core                                                                                                                                                                                                           
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                        
BuildVersion                   3.0.0.0                                                                                                                                                                                                        
GitCommitId                    v6.0.0-alpha.18                                                                                                                                                                                                
CLRVersion                                                                                                                                                                                                                                    
WSManStackVersion              3.0                                                                                                                                                                                                            
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                            
SerializationVersion           1.1.0.1                                                                                                                                                                                                        


PS /home/rw> uname -a                                                                                                                                                                                                                          
Linux rafiki 4.9.0-2-amd64 #1 SMP Debian 4.9.13-1 (2017-02-27) x86_64 GNU/Linux
~~~

**Comment by quonic** [comment]
@thrnio Are you sure that you installed the 16.04 package? I ask because libicu55 isn't available for Debian 9.

Anyways I captured an strace of powershell segfaulting after running update-help.
[powershell.txt](https://github.com/PowerShell/PowerShell/files/959690/powershell.txt)

From a Debian 9 fully updated, installed powershell_6.0.0-alpha.18-1ubuntu1.14.04.1_amd64.deb . Running the same kernel as @thrnio 

I don't know if that helps at all or not.

**Comment by thrnio** [comment]
@quonic yeah, `powershell_6.0.0-alpha.18-1ubuntu1.16.04.1_amd64.deb`. I have `libicu55` on my machine apparently, not really sure when or where it came from.

**Comment by quonic** [comment]
@thrnio if you install strace and run `strace -o ~/powershell-strace.txt powershell` then run `update-help`. That should create a file that will show what system call caused the segfault.

Short of running a debugger on powershell it self, this should help MSFT narrow down what the cause it.

**Comment by anmenaga** [comment]
@thrnio Looks like PS package for Ubuntu 16.04 (`powershell_6.0.0-alpha.18-1ubuntu1.16.04.1_amd64.deb`) has `libicu55` as dependency:

![deblibicu55](https://cloud.githubusercontent.com/assets/11860095/25459999/0b9eb7ae-2a96-11e7-930f-b9aae42bab7a.png)

**Comment by thrnio** [comment]
@anmenaga I know, I have the package installed on my system despite it not being in the Debian 9 repositories, probably because this installation is old enough that the version shipped in Debian 9 has changed.

**Comment by joeyaiello** [comment]
Setting this as HighPriority so that we at least try to get a valid repro. If this is happening on all Debian 8 machines, we need to fix it (though I don't think we care about Debian 9 *yet*).

**Comment by iSazonov** [comment]
The Issue can be resolved by #3961

**Comment by SteveL-MSFT** [comment]
Tried beta.7 on Debian8 (Debian 3.16.43) and `Find-Module` worked fine
